pages/chaoss/visualizations/contribs_by_action.py

```diff
@@ -0,0 +1,229 @@
+from queries.contributors_query import contributors_query as ctq
+import io
+from cache_manager.cache_manager import CacheManager as cm
+from pages.utils.job_utils import nodata_graph
+import time
+
+
+PAGE = "chaoss"
+VIZ_ID = "contribs-by-action"
+
+gc_contribs_by_action = dbc.Card(
+    [
+        dbc.CardBody(
+            [
+                html.H3(
+                    "Contributors by Action Type",
+                    className="card-title",
+                    style={"textAlign": "center"},
+                ),
+                dbc.Popover(
+                    [
+                        dbc.PopoverHeader("Graph Info:"),
+                        dbc.PopoverBody(
+                            """Visualizes the number of contributors in the specified time bucket that\n
+                            preformed the action selected."""
+                        ),
+                    ],
+                    id=f"popover-{PAGE}-{VIZ_ID}",
+                    target=f"popover-target-{PAGE}-{VIZ_ID}",
+                    placement="top",
+                    is_open=False,
+                ),
+                dcc.Loading(
+                    dcc.Graph(id=f"{PAGE}-{VIZ_ID}"),
+                ),
+                dbc.Form(
+                    [
+                        dbc.Row(
+                            [
+                                dbc.Label(
+                                    "Action Type:",
+                                    html_for=f"action-dropdown-{PAGE}-{VIZ_ID}",
+                                    width="auto",
+                                ),
+                                dbc.Col(
+                                    [
+                                        dcc.Dropdown(
+                                            id=f"action-dropdown-{PAGE}-{VIZ_ID}",
+                                            options=[
+                                                {
+                                                    "label": "PR Open",
+                                                    "value": "PR Open",
+                                                },
+                                                {"label": "Comment", "value": "Comment"},
+                                                {"label": "PR Review", "value": "PR Review"},
+                                                {"label": "Issue Opened", "value": "Issue Opened"},
+                                                {"label": "Issue Closed", "value": "Issue Closed"},
+                                                {"label": "Commit", "value": "Commit"},
+                                            ],
+                                            value="PR Open",
+                                            clearable=False,
+                                        ),
+                                        dbc.Alert(
+                                            children="Please select different action as this set has no contributions of this type'",
```

Comment:
Maybe a slight rewording:
"No contributions of this type have been made. Please select a different contribution type."